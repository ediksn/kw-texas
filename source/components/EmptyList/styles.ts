import { StyleSheet } from 'react-native'
import { verticalScale, moderateScale } from 'react-native-size-matters'
import { theme } from '~/constants'
import { useDeviceWidth } from '~/hooks'

const DEVICE_WIDTH = useDeviceWidth()

export const styles = StyleSheet.create({
  container: {
    justifyContent: 'center',
    alignItems: 'center',
    flex: 1
  },
  containerTitles: {
    width: DEVICE_WIDTH - 96,
    alignItems: 'center'
  },
  icon: {
    width: moderateScale(60),
    height: moderateScale(60),
    marginBottom: verticalScale(40)
  },
  title: {
    fontSize: theme.fonts.LARGE_SIZE,
    fontFamily: 'Mulish-Bold',
    color: theme.lightColor
  },
  subTitle: {
    marginTop: verticalScale(16),
    fontSize: theme.fonts.SMALL_SIZE,
    fontFamily: 'Mulish-Regular',
    color: theme.lightColor,
    textAlign: 'center'
  }
})
