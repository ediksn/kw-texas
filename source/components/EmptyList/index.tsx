import React from 'react'
import { useTranslation } from 'react-i18next'
import { Image, Text, View, ImageStyle } from 'react-native'
import { styles } from './styles'

interface Props {
  title: string
  icon: any
  iconStyle: ImageStyle
}

const EmptyList = ({ title, icon, iconStyle }: Props) => {
  const { t } = useTranslation()

  return (
    <View style={styles.container}>
      <Image source={icon} style={iconStyle} />
      <Text style={styles.text}>{t(title)}</Text>
    </View>
  )
}

export default EmptyList
