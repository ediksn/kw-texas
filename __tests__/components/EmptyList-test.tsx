import React from 'react'
import renderer from 'react-test-renderer'
import EmptyList from '~/components/EmptyList'

jest.mock('react-i18next', () => ({
  useTranslation: () => ({ t: (key: any) => key })
}))

test('renders correctly', () => {
  const title = 'Title'
  const icon = null
  const iconStyle = {}
  const emtpyList = renderer.create(<EmptyList title={title} icon={icon} iconStyle={iconStyle} />).toJSON()
  expect(emtpyList).toMatchSnapshot()
})
