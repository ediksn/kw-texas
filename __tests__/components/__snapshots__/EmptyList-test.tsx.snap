// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "justifyContent": "center",
    }
  }
>
  <View
    style={
      Object {
        "alignItems": "center",
        "width": 654,
      }
    }
  >
    <Image
      source={null}
      style={
        Object {
          "height": 94.28571428571428,
          "marginBottom": 78.47058823529412,
          "width": 94.28571428571428,
        }
      }
    />
    <Text
      style={
        Object {
          "color": "#3D424D",
          "fontFamily": "Mulish-Bold",
          "fontSize": 25.142857142857142,
          "textAlign": "center",
        }
      }
    >
      Title
    </Text>
  </View>
</View>
`;
